precompile/modules/initia_stdlib/sources/dex.move

```diff
@@ -2384,4 +2506,125 @@ module initia_std::dex {
+        let usdc_metadata = coin::metadata(chain_addr, string::utf8(b"USDC"));
+
+        coin::mint_to(&initia_mint_cap, chain_addr, 100000000);
+        coin::mint_to(&usdc_mint_cap, chain_addr, 100000000);
+
+        // spot price is 1
+        create_pair_script(
+            chain,
+            std::string::utf8(b"name"),
+            std::string::utf8(b"SYMBOL"),
+            bigdecimal::from_ratio_u64(3, 1000),
+            bigdecimal::from_ratio_u64(8, 10),
+            bigdecimal::from_ratio_u64(2, 10),
+            coin::metadata(chain_addr, string::utf8(b"INIT")),
+            coin::metadata(chain_addr, string::utf8(b"USDC")),
+            80000000,
+            20000000
+        );
+
+        assert!(
+            coin::balance(chain_addr, init_metadata) == 20000000,
+            0
+        );
+        assert!(
+            coin::balance(chain_addr, usdc_metadata) == 80000000,
+            1
+        );
+
+        let lp_metadata = coin::metadata(chain_addr, string::utf8(b"SYMBOL"));
+        assert!(
+            coin::balance(chain_addr, lp_metadata) == 80000000,
+            2
+        );
+
+        move_to(
+            chain,
+            CoinCapsInit {
+                burn_cap: initia_burn_cap,
+                freeze_cap: initia_freeze_cap,
+                mint_cap: initia_mint_cap
+            }
+        );
+
+        move_to(
+            chain,
+            CoinCapsUsdc {
+                burn_cap: usdc_burn_cap,
+                freeze_cap: usdc_freeze_cap,
+                mint_cap: usdc_mint_cap
+            }
+        );
+    }
+
+    #[test(chain = @0x1, borrower = @0x1782)]
+    fun test_dex_flash_swap(
+        chain: &signer, borrower: &signer
+    ) acquires ModuleStore, Pool, CoinCapabilities, Config, FlashSwap, CoinCapsInit {
+        test_setup(chain);
+        let chain_addr = signer::address_of(chain);
+        let borrower_addr = signer::address_of(borrower);
+
+        let init_metadata = coin::metadata(chain_addr, string::utf8(b"INIT"));
+        let lp_metadata = coin::metadata(chain_addr, string::utf8(b"SYMBOL"));
+        let pair = object::convert<Metadata, Config>(lp_metadata);
+        let pair_addr = object::object_address(&pair);
+
+        // flash_swap init to usdc
+        let (return_fa, loan) = flash_swap(pair, init_metadata, 1000, option::none());
+        assert!(fungible_asset::amount(&return_fa) == 996, 3);
+        assert!(exists<FlashSwap>(pair_addr), 4);
+        coin::deposit(borrower_addr, return_fa);
+
+        // repay INIT
+        coin::mint_to(
+            &borrow_global<CoinCapsInit>(chain_addr).mint_cap, borrower_addr, 1000
+        );
+        let offer_fa = coin::withdraw(borrower, init_metadata, 1000);
+        repay_flash_swap(offer_fa, loan);
+        assert!(!exists<FlashSwap>(pair_addr), 5);
+    }
+
+    #[test(chain = @0x1, borrower = @0x1782)]
+    #[expected_failure(abort_code = 0x10017, location = Self)]
+    fun test_dex_flash_swap_not_enough_repayment(
```

Comment:
it also aborts when borrower repays too much. wonder if we also want to add a test case to cover it